src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java

```diff
@@ -352,7 +355,35 @@ private boolean sendMail(ExtendedEmailPublisherContext context) {
 
                     while (true) {
                         try {
-                            Transport.send(msg);
+                            ExtendedEmailPublisherDescriptor descriptor = getDescriptor();
+                            String smtpHost = descriptor.getSmtpServer();
+                            if (smtpHost == null) {
+                                smtpHost = "";
+                            }
+                            String awsRegex = "^email-smtp\\.([a-z0-9-]+)\\.amazonaws\\.com$";
+                            Pattern p = Pattern.compile(awsRegex);
+                            Matcher m = p.matcher(smtpHost);
+                            if (m.matches()) {
+                                String region = m.group(1);
+                                Session session = descriptor.createSession();
+                                Transport transport = session.getTransport(allRecipients[0]);
+                                try {
+                                    transport.connect();
+                                    transport.sendMessage(msg, allRecipients);
+                                    if (transport instanceof SMTPTransport) {
+                                        String response = ((SMTPTransport)transport).getLastServerResponse();
+                                        String[] parts = response.trim().split(" +");
+                                        if (parts.length == 3 && parts[0].equals("250") && parts[1].equals("Ok")) {
+                                            String MessageID = "<" + parts[2] + "@" + region + ".amazonses.com>";
+                                            msg.setHeader("Message-ID", MessageID);
+                                        }
```

Comment:
It is not possible, because the new message id is only available after the email has been sent, not before.

If I put the code that sends the actual email in the pre-send script, the email will be sent twice, and if I set `cancel = true`, the message id is lost for the In-Reply-Header.